clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -8522,6 +8522,22 @@ bool Sema::CheckCountedByAttr(Scope *S, const FieldDecl *FD) {
     }
   }
 
+  // We don't support 'counted_by' on flexible array members in substructures.
+  const DeclContext *DC = FD->getParent();
+  while (const auto *RD = dyn_cast<RecordDecl>(DC)) {
+    if (!RD->isAnonymousStructOrUnion() ||
+        !isa<RecordDecl>(RD->getLexicalParent()))
+      break;
+    DC = RD->getLexicalParent();
+  }
```

Comment:
```
while (true) {
  const auto *RD = dyn_cast<RecordDecl>(DC);
  if (!RD || !RD->isAnonymousStructOrUnion())
    break;
  DC = RD->getLexicalParent();
}
```

Reply:
I'll do it, but could you explain why this is better? Is it a style guide issue?

Reply:
It's computing and casting things once instead of twice.  It's just generally much clearer.

Reply:
Not really. I'm checking to make sure that the `RecordDecl`'s lexical parent is also a `RecordDecl`. I need to do this, otherwise `DC` will end up as `TranslationUnit` in some cases.